src/pages/registration.tsx

```diff
@@ -14,64 +21,513 @@ const Container = styled.div`
+            type="text"
+            name="question1"
+            value={userForm.question1.value}
+            placeHolder="What do you to learn at cuHacking 2021: Snowed In? (500 characters max)"
+            displayLabel
+            padded
+            style={{ width: "100%" }}
+            maxLength={500}
+          ></TextArea>
+        </FormRow>
+        <FormRow>
+          <TextArea
+            onChange={handleFormChange}
+            type="text"
+            name="question2"
+            value={userForm.question2.value}
+            placeHolder="What are you passionate about? What are you interested in? (500 characters max)"
+            displayLabel
+            padded
+            style={{ width: "100%" }}
+            maxLength={500}
+          ></TextArea>
+        </FormRow>
+        <FormRow>
+          <TextArea
+            onChange={handleFormChange}
+            type="text"
+            name="question3"
+            value={userForm.question3.value}
+            placeHolder="Describe a big milestone you've recently achieved (500 characters max)"
+            displayLabel
+            padded
+            style={{ width: "100%" }}
+            maxLength={500}
+          ></TextArea>
+        </FormRow>
+      </Form>
     </FormContainer>
   );
 
   const currentStep = () => {
-    switch(step) {
-      case 0: return personalInformation;
-    };
+    switch (step) {
+      case steps.personalInformation:
+        return personalInformation;
+      case steps.education:
+        return education;
+      case steps.experience:
+        return experience;
+      case steps.additional:
+        return additional;
+      case steps.questions:
+        return questions;
+    }
   };
- 
+
+  const previousStep = () => {
+    if (step === steps.personalInformation) return;
+    setStep(step - 1);
+  };
+
+  const nextStep = () => {
+    if (step === finalStep) {
+      console.log("verify the form");
+      console.log("send request to API");
+      return;
+    }
+    setStep(step + 1);
+  };
+
   const actions = (
     <InteractionsContainer>
-      <Button />
-      <Button />
+      <Button role="secondary" onClick={() => previousStep()}>
+        <Desktop>
+          {step === steps.personalInformation ? "cancel" : "back"}
+        </Desktop>
+        <Mobile>
+          <LeftArrow src={darkArrow}></LeftArrow>
+        </Mobile>
+      </Button>
+      <MarkersContainer>
+        {[...Array(SECTIONS)].map((_, index) =>
+          step < index ? (
+            <StepMarker key={index}></StepMarker>
+          ) : (
+            <StepMarkerDone key={index}></StepMarkerDone>
+          )
+        )}
+      </MarkersContainer>
+      <Button onClick={() => nextStep()}>
+        <Desktop>{step === steps.additional ? "submit" : "continue"}</Desktop>
+        <Mobile>
+          <RightArrow
+            src={globalTheme === "dark" ? lightArrow : darkArrow}
+          ></RightArrow>
+        </Mobile>
+      </Button>
```

Comment:
```suggestion
      <Button role="secondary" onClick={() => previousStep()}>
        <Desktop>
          {step === steps.personalInformation ? "cancel" : "back"}
        </Desktop>
        <Mobile>
          <LeftArrow alt="..." src={darkArrow}></LeftArrow>
        </Mobile>
      </Button>
      <MarkersContainer>
        {[...Array(SECTIONS)].map((_, index) =>
          step < index ? (
            <StepMarker key={index}></StepMarker>
          ) : (
            <StepMarkerDone key={index}></StepMarkerDone>
          )
        )}
      </MarkersContainer>
      <Button onClick={() => nextStep()}>
        <Desktop>{step === steps.additional ? "submit" : "continue"}</Desktop>
        <Mobile>
          <RightArrow alt="..."
            src={globalTheme === "dark" ? lightArrow : darkArrow}
          ></RightArrow>
        </Mobile>
      </Button>
```

These images need `alt` text.

Reply:
Done